test/images/README.md

```diff
@@ -0,0 +1,117 @@
+new images, test the changes made.
+
+
+## Prerequisites
+
+In order to build the docker test images, a Linux node is required. The node will require `make`
+and `docker (version 18.06.0 or newer)`. Manifest lists were introduced in 18.03.0, but 18.06.0
+is recommended in order to avoid certain issues.
+
+The node must be able to push the images to the desired container registry. Depending on the
+container registry, it might require a different authentication method. For dockerhub, this
+can be done by running the command:
+
+```bash
+    docker login -u your-awesome-username -p anAwesomerPassword
+```
+
+
+## Making changes to images
+
+There are several thousands of tests in Kubernetes E2E testing. Not all of them are being run on
+new PRs, and thus, not all images are used, especially those that are not used by Conformance tests.
+
+So, in order to prevent regressions in the images and failing jobs, any changes made to the image
+itself or its binaries will require the image's version to be bumped. In the case of a regression
+which cannot be immediately resolved, the image version used in E2E tests will be reverted to the
+last known stable version.
+
+The version can easily be bumped by modifying the file `test/images/IMAGE_NAME/VERSION`, which will
+be used when building the image. Additionally, for the `agnhost` image, also bump the `Version` in
+`test/images/agnhost/agnhost.go`.
+
+The typical image used in E2E testing is the `agnhost` image. It contains several subcommands with
+different [functionalities](agnhost/README.md), used to validate different Kubernetes behaviours. If
+a new functionality needs testing, consider adding an `agnhost` subcommand for it first, before
+creating an entirely separate test image.
+
+Some test images (`mounttest`, `test-webserver`) are used as bases for other images (`mounttest-user`,
+`kitten`, `nautilus`). If the parent image's `VERSION` has been bumped, also bump the version in the
+children's `BASEIMAGE` files in order for base image changes to be reflected in the child images as well.
+
+
+## Building images
+
+The images are built through `make`. Since some images (`mounttest`, `test-webserver`)
+are used as a base for other images, it is recommended to build them first, if needed.
+
+An image can be built by simply running the command:
+
+```bash
+    make all WHAT=test-webserver
+```
+
+To build AND push an image, the following command can be used:
+
+```bash
+    make all-push WHAT=test-webserver
+```
+
+By default, the images will be tagged and pushed under the `gcr.io/kubernetes-e2e-test-images`
+registry. That can changed by running this command instead:
+
+```bash
+    REGISTRY=foo_registry make all-push WHAT=test-webserver
+```
+
+*NOTE* (for test `gcr.io` image publishers): Some tests (e.g.: `should serve a basic image on each replica with a private image`)
+require the `agnhost` image to be published in an authenticated repo as well:
```

Comment:
I think we should find a way to rewrite this test to use an image that is updated less frequently than agnhost for this test so we don't have to push to two repos

Reply:
We could. The test basically uses `/agnhost serve-hostname`, and checks the hostname of every pod in the replica set by proxying into the pod and doing a GET.

We could just use the other images in the authenticated registry (alpine:3.7, windows-nanoserver:v1) and exec ``hostname`` into the pods to get the same results. This will have to be done in another PR IMO. Thoughts?